src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -0,0 +1,255 @@
+package org.broadinstitute.hellbender.utils.read;
+
+import com.google.api.services.genomics.model.Read;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.util.Locatable;
+
+import java.util.UUID;
+
+/**
+ * Unified read interface for use throughout the GATK.
+ *
+ * This is the read-only subset of possible operations. Use {@link MutableGATKRead} if you need to modify
+ * reads in-place.
+ */
+public interface GATKRead extends Locatable {
+
+    /**
+     * @return A globally unique identifier for this read, suitable for use as a key
```

Comment:
what does 'globally' mean? Anywhere in the world or just in this dataflow worker? Ie, is it like a hashcode or it this unique across machines?